exercises/wordy/src/example/java/WordProblemSolver.java

```diff
@@ -0,0 +1,60 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+final class WordProblemSolver {
+
+    private static final String INTEGER_REGEX_STRING = "(-?\\d+)";
+
+    private static final String BINARY_OPERATION_REGEX_STRING
+        = "(plus|minus|multiplied by|divided by|raised to the power)";
+
+    private static final String VALID_QUESTION_REGEX_STRING =
+            "^What is " + INTEGER_REGEX_STRING +
+            "( " + BINARY_OPERATION_REGEX_STRING + " " + INTEGER_REGEX_STRING + ")+" +
+            "\\?$";
+
+    int solve(final String wordProblem) {
+        if (!wordProblem.matches(VALID_QUESTION_REGEX_STRING)) {
+            throw new IllegalArgumentException("I'm sorry, I don't understand the question!");
+        }
+
+        final Matcher initialValueMatcher = Pattern.compile(INTEGER_REGEX_STRING).matcher(wordProblem);
+
+        initialValueMatcher.find();
+        int result = Integer.parseInt(initialValueMatcher.group());
+
+        final Matcher operationAndValueMatcher
+                = Pattern.compile(BINARY_OPERATION_REGEX_STRING + " " + INTEGER_REGEX_STRING).matcher(wordProblem);
+
+        while (operationAndValueMatcher.find()) {
+            result = applyBinaryOperation(
+                result,
+                operationAndValueMatcher.group(1),
+                Integer.parseInt(operationAndValueMatcher.group(2)));
+        }
+
+        return result;
+    }
+
+    private int applyBinaryOperation(
+            final int firstValue,
+            final String operationDescription,
+            final int secondValue) {
+
+        switch (operationDescription) {
+            case "plus":
+                return firstValue + secondValue;
+            case "minus":
+                return firstValue - secondValue;
+            case "multiplied by":
+                return firstValue * secondValue;
+            case "divided by":
+                return firstValue / secondValue;
+            case "raised to the power":
```

Comment:
I'm not sure I clearly described what I had hoped to here. Let me try again:

In the canonical tests, exponentiation is included as extra credit based on the fact that supporting a sentence structure like "What is 52 cubed?" would require additional sentence structure matching. However, since I noted that "What is 52 raised to the power 3?" fit the existing structure, I wrote a test for that case which _doesn't_ require new matching code. This felt opportunistic but is a deviation from the canonical data and its intent.

@jtigger does that perturb you more? I think it should.